helix-core/src/lib.rs

```diff
@@ -92,6 +92,19 @@ pub fn config_dir() -> std::path::PathBuf {
     path
 }
 
+/// First looks for the local `.helix` directory in CWD. If it can't find it, then
+/// it searches for it within the root `.git` directory. Otherwise, it uses the CWD
+/// to avoid an unwrap.
+pub fn local_config_dir() -> std::path::PathBuf {
+    let cwd = std::env::current_dir().expect("unable to determine current directory");
+    if cwd.join(".helix").is_dir() {
+        cwd.join(".helix")
+    } else {
+        let root = find_root(None).unwrap_or(cwd);
+        root.join(".helix")
+    }
+}
```

Comment:
Wouldn't user to expect the local to overwrite root config if both exists? Shouldn't we read both and merge them with local being the last to be merged?